dpc-portal/app/controllers/invitations_controller.rb

```
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+# Handles acceptance of invitations
+class InvitationsController < ApplicationController
+  before_action :load_organization
+  before_action :load_invitation
+  before_action :authenticate_user!, except: %i[login]
+  before_action :invitation_matches_cd, only: %i[confirm]
+
+  def accept
+    if current_user.email != @cd_invitation.invited_email
+      return render(Page::CredentialDelegate::BadInvitationComponent.new('pii_mismatch'),
+                    status: :forbidden)
+    end
+
+    render(Page::CredentialDelegate::AcceptInvitationComponent.new(@organization, @cd_invitation))
+  end
+
+  def confirm
+    CdOrgLink.create!(user: current_user, provider_organization: @organization, invitation: @cd_invitation)
+    @cd_invitation.update!(invited_given_name: nil, invited_family_name: nil, invited_phone: nil, invited_email: nil)
+    flash[:notice] = "Invitation accepted. You can now manage this organization's credentials. Learn more."
+    redirect_to organizations_path
+  end
+
+  def login
+    login_session
+    client_id = "urn:gov:cms:openidconnect.profiles:sp:sso:cms:dpc:#{ENV.fetch('ENV', nil)}"
+    url = URI::HTTPS.build(host: 'idp.int.identitysandbox.gov',
+                           path: '/openid_connect/authorize',
+                           query: { acr_values: 'http://idmanagement.gov/ns/assurance/ial/2',
+                                    client_id:,
```

Probably should throw an error on the `nil` case, to avoid sending an invalid request.

---

Throws!